### ISSUE TITLE
Infer the exposed key from the key of the field?

### ISSUE THREAD

**Issue by mmahalwy** [hayes/giraphql#54]
This is probably a feature request.

Is it possible to infer the exposed key from the key of the field? So instead of this:
```ts
const ApplicationObjectType = builder.objectRef<Application>('Application').implement({
  fields: (t) => ({
    applicationFormConfigId: t.exposeID('applicationFormConfigId', {}),
    clientId: t.exposeID('clientId', {}),
    comments: t.expose('comments', { type: 'GraphQLJSON' }),
    destinations: t.exposeStringList('destinations', {}),
...
```

It'd be this
```ts
const ApplicationObjectType = builder.objectRef<Application>('Application').implement({
  fields: (t) => ({
    applicationFormConfigId: t.exposeID(), // inferred as `applicationFormConfigId`
    clientId: t.exposeID(), // inferred as `clientId`
    comments: t.expose({ type: 'GraphQLJSON' }),
    destinations: t.exposeStringList(),
...
```

**Comment by hayes** [comment]
Unfortunately this is not possible due to limitations with typescripts inference.  I spent a lot of time looking at this when first building out GiraphQL. Current API is at least explicit and makes it easy to do aliasing.

**Comment by hayes** [comment]
Long term, what I may do is add an API like Prisma where you don't need to return the fields in an object.   

`define: t => { t.exposeID('clientID', options) }`.  Core of GiraphQL already has most of the pieces to make this work.  I would probably initially add this as a Plugin.  The need to return fields at all is a bit vestigial from early versions that worked more like the simple objects plugin.  It allowed some cool stuff around interfaces, but keeping the types correct without getting in the way too much proved to be too complicated. 

Long story short, I agree the current flow isn't perfect, and I have a few ideas on API changes that might help, but they are non-trivial and might be a long way out.

**Comment by mmahalwy** [comment]
Totally understand where you're coming from. I actually like returning the object because it's easier to read instead of scanning the code for what comes after `t.*(`.

**Comment by hayes** [comment]
going to close this since there isn't really anything actionable here in the short term.
